applicationinsights/channel/AsynchronousSender.py

```diff
@@ -110,6 +110,12 @@ def _run(self):
             local_send_time = local_send_interval
         while True:
             while True:
+                # Don't even bother processing if buffer is less than 100 items (the default batch size)
+                # This prevents us from wasting time sending tiny batches of data
+                # fixme magic number
+                if local_queue._queue.qsize() < 100:
```

Comment:
Were you able to get a feel for how delayed this has to be to keep performance up? I'm ok with a 3 second delay if it has to be that long, but something like 300-500ms would be better.